src/bench/disconnected_transactions.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <test/util/random.h>
+#include <test/util/setup_common.h>
+#include <validation.h>
+
+constexpr size_t BLOCK_VTX_COUNT{4000};
+constexpr size_t BLOCK_VTX_COUNT_10PERCENT{400};
+
+using BlockTxns = decltype(CBlock::vtx);
+
+/** Reorg where 1 block is disconnected and 2 blocks are connected. */
+struct ReorgTxns {
+    /** Disconnected block. */
+    BlockTxns disconnected_txns;
+    /** First connected block. */
+    BlockTxns connected_txns_1;
+    /** Second connected block, new chain tip. Has no overlap with disconnected_txns. */
+    BlockTxns connected_txns_2;
+    /** Transactions shared between disconnected_txns and connected_txns_1. */
+    size_t num_shared;
+};
+
+static BlockTxns CreateRandomTransactions(size_t num_txns, uint32_t unique_set_num)
+{
+    FastRandomContext det_rand{true};
+    // The unique_set_num trick won't work otherwise. It assumes that there's a different
+    // unique_set_num each time we call CreateRandomTransactions and each batch is <= 4000.
+    assert(num_txns <= BLOCK_VTX_COUNT);
+
+    BlockTxns txns;
+    txns.reserve(num_txns);
+    // Simplest spk for every tx
+    CScript spk = CScript() << OP_TRUE;
+    for (uint32_t i = 0; i < num_txns; ++i) {
+        CMutableTransaction tx;
+        tx.vin.resize(1);
+        // We should get a different prevout hash every time. But just to be sure, change the index
+        // as well to ensure every tx has a different txid.
+        tx.vin.emplace_back(CTxIn{COutPoint{det_rand.rand256(), uint32_t(i + BLOCK_VTX_COUNT * unique_set_num)}});
```

Comment:
Good point about using a static var. I've simplified this but instead of using locktime or prevout `n`, I'm just having each tx spend the previous one. Should be sufficient to ensure we don't get any duplicates, so that gets rid of unique_set_num and the rng.